samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -1,276 +1,215 @@
+            "  Action:\n"
+            "   {%- for action in action_list %}\n"
+            "     # - {{action}}\n"
+            "   {%- endfor %}\n"
+            "  Resource: {{arn}}\n"
         )
 
-    if len(resource_tag_mapping_list) == 0:
-        raise KeyError(f"No resources match the tag: {tag_filters}, cannot generate any IAM permissions.")
-
-    iam_statements = [_create_iam_statment_string(resource["ResourceARN"]) for resource in resource_tag_mapping_list]
-    return "".join(iam_statements)
-
-
-def generate_test_runner_template_string(
-    boto_client_provider: BotoProviderType,
-    jinja_base_template: str,
-    s3_bucket_name: str,
-    image_uri: str,
-    tag_filters: Union[dict, None],
-) -> Union[dict, None]:
-    """
-    Renders a base jinja template with a set of parameters needed to create the Test Runner Stack.
-
-    Parameters
-    ----------
-
-    boto_client_provider : BotoProviderType
-        Provides a boto3 client in order to query tagging API
-
-    jinja_base_template : str
-        The jinja YAML template to which parameters are substituted.
-
-    s3_bucket_name : str
-        The name of the S3 bucket used by the Test Runner Stack.
+        for arn_regex, action_list in default_permissions_map.items():
+            if re.search(arn_regex, resource_arn) is not None:
+                LOG.debug("Matched ARN `%s` to IAM actions %s", resource_arn, action_list)
+
+                # APIGW API IAM Statements:
+                # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+                if arn_regex in (APIGW_API_REGEX, APIGW_RESTAPI_REGEX):
+                    apiId = self._extract_api_id_from_arn(resource_arn)
+                    execute_api_arn = (
+                        f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/*/GET/*"
+                    )
+                    return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+                        action_list=action_list, arn=execute_api_arn
+                    )
+                else:
+                    return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+                        action_list=action_list, arn=resource_arn
+                    )
+
+        # If the supplied ARN is for a resource without any default actions supported
+        LOG.debug("Found no match for ARN `%s` to any IAM actions.", resource_arn)
+        return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
+            action_list=[], arn=resource_arn
+        )
 
-    image_uri : str
-        The URI of the Image to be used by the Test Runner Fargate task definition.
+    def _get_default_bucket_name(self) -> str:
+        """
+        Returns a guaranteed unique default S3 Bucket name
+        """
+        return "test-runner-bucket" + str(uuid.uuid4())
```

Comment:
Minor nit; consider a `-` separator between 'bucket' and the uuid.

Reply:
Will do, should be there.